File: Projekt/Backend/system-kliniki-medycznej/src/main/java/org/back/systemklinikimedycznej/appointment/controllers/dto/AppointmentDetails.java
package org.back.systemklinikimedycznej.appointment.controllers.dto;

import lombok.Builder;
import org.back.systemklinikimedycznej.doctor.controller.dto.DoctorInfo;
import org.back.systemklinikimedycznej.patient.controllers.dto.PatientData;
import org.back.systemklinikimedycznej.patient.repositories.entities.patient_card.PatientCard;

import java.util.UUID;

@Builder
public record AppointmentDetails(
        AppointmentInfo appointment,
        PatientData patientData,
        DoctorInfo doctor,
        UUID patientCardId
) {
}

File: Projekt/Frontend/system-kliniki-medycznej/src/app/features/appointment/model/appointment-details.ts
import {AppointmentInfo} from './appointment-info';
import {PatientData} from '../../patient/model/patient-data';
import {DoctorInfo} from '../../doctor/domain/doctor-info';

export interface AppointmentDetails {
  appointment: AppointmentInfo;
  patientData: PatientData;
  doctor: DoctorInfo;
  patientCardId: string;
}
